static/js/stream_create.js

```diff
@@ -116,48 +116,71 @@ class StreamNameError {
 const stream_name_error = new StreamNameError();
 
 // Stores the previous state of the stream creation checkbox.
-let stream_announce_previous_value =
+let announce_stream_previous_value =
     settings_data.user_can_create_public_streams() ||
     settings_data.user_can_create_web_public_streams();
+let default_stream_previous_value = false;
 
-// Within the new stream modal...
-function update_announce_stream_state() {
-    // If there is no notifications_stream, we simply hide the widget.
-    if (!stream_data.realm_has_notifications_stream()) {
-        $("#announce-new-stream").hide();
-        return;
-    }
-
-    // If the stream is invite only, disable the "Announce stream" option.
-    // Otherwise enable it.
-    const $announce_stream_checkbox = $("#announce-new-stream input");
-    const $announce_stream_label = $("#announce-new-stream");
-    let disable_it = false;
+function update_state($elem_label, elem_previous_value) {
```

Comment:
We can have a better name here. Something like `update_setting_state_based_on_stream_privacy`.